sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -1198,8 +1209,34 @@ def process(self, element, *schema_side_inputs):
       return self._flush_all_batches()
 
   def finish_bundle(self):
+    current_millis = int(time.time() * 1000)
```

Comment:
I noticed that you have three calls to time.time(). This is an expensive system call. Is there any way to eliminate this additional usage here? I.e. You could just update the current_millis, in other places where you call time.time().

Though since this is in finish_bundle, maybe this is fine
Though see other comment, not sure its wise to do this in finish_bundle. We still need to be careful that we don't starve out the logging. The solution to that may be to log in both finish bundle (always). and after each request (if we have exceeded the logging interval)
@chamikaramj WDYT?

Reply:
The three calls have their own role. One for remembering request time. One for measuring response time so that we can subtract the start time from the end time. And the last one is for checking log reporting interval.

We might slightly reduce the overhead by moving the last `time.time()` call inside the synchronized block. In that case, the thread that skips the synchronized section won't execute `time.time()` (probably the saving is small though)